### DISCUSSION TITLE
Replace `--no-user-output` with `--output-format none` (or similar)

### DISCUSSION THREAD

**Discussion by Marcono1234** [foresterre/cargo-msrv#970] [General]
Currently there are two command line arguments to control the output format:
- `--no-user-output`
- `--output-format <format>`

This creates the following issues:
- It is _syntactically_ valid to combine both arguments: `cargo msrv verify --no-user-output --output-format human`
But semantically this makes no sense and is rather confusing.
- The struct modelling the output configuration similarly allows both values to be set:
https://github.com/foresterre/cargo-msrv/blob/ab2870f41a8bace2cba5c63a3dfa17e534e4a903/src/cli/shared_opts.rs#L27-L41
- There is custom conversion code needed
https://github.com/foresterre/cargo-msrv/blob/ab2870f41a8bace2cba5c63a3dfa17e534e4a903/src/context.rs#L316-L328

In total this makes usage a bit confusing, and the implementation rather complex. What do you think about replacing `--no-user-output` with a new `none` output format (or similar)? For example: `--output-format none`

This should hopefully solve the issues mentioned above.

**Comment by foresterre** [comment]
I originally added it because I felt the `--no-{}` signalled that something was not enabled (a boolean). For this case there would be two options: "print user output" (_enabled_) and "do not print user output" (_disabled_).
It is true that I only provided one flag (for disabled), because it is the opposite of the default (enabled).

The _type_ of user output I reasoned is something different. If user output is enabled, then we can still print json output, or something a little more human readable.

**Reply by foresterre** [reply]
I suppose I could have coded this difference more explicitly by making the output format `Option<OutputFormat>` (although the default should be `Some(OutputFormat::Human)`, instead of the default of `Option` which is `None`.

**Reply by foresterre** [reply]
Perhaps another idea could be to add a `conflicts_with` [1] method on `--no-output-format` and have it conflict with `--output-format <value>`.

[1] https://docs.rs/clap/latest/clap/struct.Arg.html#method.conflicts_with

**Reply by Marcono1234** [reply]
Yes `conflicts_with` works, thanks for the suggestion. I have created a pull request for it now: #979

Together the with `UserOutputOpts::effective_output_format()` I added in #963 this issue is in my opinion then mostly resolved.
